packages/syft/src/syft/serde/lib_service_registry.py

```diff
@@ -309,9 +341,13 @@ def __init__(self, children: List[CMPModule]):
         self.children = {c.path: c for c in children}
 
     def build(self) -> Self:
+        self.stack = []
```

Comment:
what is the benefit of doing this with a global stack and not recursive?

Reply:
in my opinion, it is cleaner to check if we have already inspected submodules in lower-level files if we have a bfs approach, rather than a recursive one, as we need to keep that information as a global variable anyway, so it gets messier if we do both